### ISSUE TITLE
WSL2 Terminal doesn't load Python venv on start

### ISSUE THREAD

**Issue by nstein-gpjoule** [microsoft/vscode#176608]
Type: <b>Bug</b>

When opening a Python project in a WSL2 enviroment in a new window, the terminal opens without activating the venv. This only happens when opening a new instance of VSCode. When the the project is already open and I open a new terminal the venv is activated.
I guess on startup the terminal is fully loaded before VSCode has detected the venv.

VS Code version: Code 1.76.0 (92da9481c0904c6adfe372c12da3b7748d74bdcb, 2023-03-01T10:22:44.506Z)
OS version: Windows_NT x64 10.0.19045
Modes:
Sandboxed: Yes
Remote OS version: Linux x64 5.15.90.1-microsoft-standard-WSL2

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|12th Gen Intel(R) Core(TM) i7-1255U (12 x 2611)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: disabled_off|
|Load (avg)|undefined|
|Memory (System)|15.71GB (3.51GB free)|
|Process Argv|--crash-reporter-id efa07db7-a6a1-49bd-bc99-4fcc63568dd1|
|Screen Reader|no|
|VM|0%|

|Item|Value|
|---|---|
|Remote|WSL: Debian|
|OS|Linux x64 5.15.90.1-microsoft-standard-WSL2|
|CPUs|12th Gen Intel(R) Core(TM) i7-1255U (12 x 2611)|
|Memory (System)|7.61GB (5.07GB free)|
|VM|0%|
</details><details><summary>Extensions (45)</summary>

Extension|Author (truncated)|Version
---|---|---
project-manager|ale|12.7.0
vsc-python-indent|Kev|1.18.0
vscode-azurevirtualmachines|ms-|0.6.4
vscode-language-pack-de|MS-|1.76.2023030809
jupyter-keymap|ms-|1.1.0
remote-containers|ms-|0.282.0
remote-ssh|ms-|0.98.0
remote-ssh-edit|ms-|0.84.0
remote-wsl|ms-|0.76.1
remote-explorer|ms-|0.2.0
vscode-icons|vsc|12.2.0
jinja|who|0.0.8
vscode-django|bat|1.10.0
githistory|don|0.6.20
python-environment-manager|don|1.0.4
python-extension-pack|don|1.7.0
html-preview-vscode|geo|0.2.5
vsc-python-indent|Kev|1.18.0
azure-dev|ms-|0.3.0
vscode-azureappservice|ms-|0.24.7
vscode-azurefunctions|ms-|1.10.3
vscode-azureresourcegroups|ms-|0.6.1
vscode-azurestaticwebapps|ms-|0.11.4
vscode-azurestorage|ms-|0.15.1
vscode-azurevirtualmachines|ms-|0.6.4
vscode-cosmosdb|ms-|0.19.3
vscode-docker|ms-|1.24.0
vscode-language-pack-de|MS-|1.76.2023030809
isort|ms-|2022.8.0
python|ms-|2023.4.0
vscode-pylance|ms-|2023.3.20
jupyter|ms-|2023.2.1000592019
jupyter-keymap|ms-|1.1.0
jupyter-renderers|ms-|1.0.15
vscode-jupyter-cell-tags|ms-|0.1.6
vscode-jupyter-slideshow|ms-|0.1.5
azure-account|ms-|0.11.3
vscode-node-azure-pack|ms-|1.1.1
color-highlight|nau|2.5.0
autodocstring|njp|0.6.1
vscode-yaml|red|1.12.1
trailing-spaces|sha|0.4.1
intellicode-api-usage-examples|Vis|0.2.7
vscodeintellicode|Vis|1.2.30
jinja|who|0.0.8

(1 theme extensions excluded)

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
vslsvsres303:30308271
pythonvspyl392:30443607
vserr242cf:30382550
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
vscoreces:30445986
pythondataviewer:30285071
vscod805:30301674
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
cmake_vspar411:30581797
vsaa593cf:30376535
pythonvs932:30410667
cppdebug:30492333
vsclangdf:30486550
c4g48928:30535728
dsvsc012cf:30540253
pynewext54:30681850
azure-dev_surveyone:30548225
pyindex848:30662994
nodejswelcome1:30587005
2e4cg342:30602488
pyind779:30671433
f6dab269:30613381
pythonsymbol12:30671437
vsctsb:30677850
vscodeenable:30660116
pythonb192cf:30669361
pythonms35cf:30680985

```

</details>

**Comment by NicolasM-Forsk** [comment]
https://github.com/microsoft/vscode-python/issues/11039

**Comment by karrtikr** [comment]
Set `"python.terminal.activateEnvInCurrentTerminal": true` to achieve the behavior you desire, this should be solved by #11039 as you pointed out.
